From 06b256b7261fe2fe9ad45f33eb9b6efcf7d8508d Mon Sep 17 00:00:00 2001
From: Christian Dreher <cdreher@prophesee.ai>
Date: Fri, 10 Nov 2023 15:04:47 +0100
Subject: [PATCH 01/13] [MV-1764] Debug: 0-init v4l2_dbg_register structs

Some implementation reject commands with inconsistent
options. This allow to perform dbg accesses on more
pendentic devices.
---
 hal_psee_plugins/src/boards/v4l2/v4l2_board_command.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/hal_psee_plugins/src/boards/v4l2/v4l2_board_command.cpp b/hal_psee_plugins/src/boards/v4l2/v4l2_board_command.cpp
index c261c1a7b0..593c0f3db7 100644
--- a/hal_psee_plugins/src/boards/v4l2/v4l2_board_command.cpp
+++ b/hal_psee_plugins/src/boards/v4l2/v4l2_board_command.cpp
@@ -117,8 +117,8 @@ unsigned int V4L2BoardCommand::get_device_count() {
 
 std::vector<uint32_t> V4L2BoardCommand::read_device_register(uint32_t device, uint32_t address, int nval) {
     std::vector<uint32_t> res;
-    struct v4l2_dbg_match match;
-    struct v4l2_dbg_register get_reg;
+    struct v4l2_dbg_match match      = {0};
+    struct v4l2_dbg_register get_reg = {0};
     int i, retval;
 
     match.type    = V4L2_CHIP_MATCH_BRIDGE;
@@ -137,8 +137,8 @@ std::vector<uint32_t> V4L2BoardCommand::read_device_register(uint32_t device, ui
 }
 
 void V4L2BoardCommand::write_device_register(uint32_t device, uint32_t address, const std::vector<uint32_t> &val) {
-    struct v4l2_dbg_match match;
-    struct v4l2_dbg_register set_reg;
+    struct v4l2_dbg_match match      = {0};
+    struct v4l2_dbg_register set_reg = {0};
     int i, retval;
 
     match.type    = V4L2_CHIP_MATCH_BRIDGE;
-- 
2.39.2

